src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/repositories/impl/HBaseRepositoryImpl.kt

```diff
@@ -30,11 +31,21 @@ class HBaseRepositoryImpl(
         topicName: String,
         records: List<UnreconciledRecord>
     ): List<Pair<UnreconciledRecord, Boolean>> {
+
+        val replicaId = if (replicationFactor == -1) -1 else randomiseReplicaId(replicationFactor)
```

Comment:
you already have the logic for returning a default region in the `randomiseReplicaId` function, so no need to duplicate
```suggestion
        val replicaId = randomiseReplicaId(replicationFactor)
```